ts/test/test-init-config.ts

```diff
@@ -70,8 +82,10 @@ describe('initConfig', () => {
       zone: 'zone',
       projectId: 'fake-projectId'
     };
-    const initializedConfig = await initConfig(config);
-    assert.deepEqual(initializedConfig, extend(config, internalConfigParams));
+    const profiler: Profiler = await createProfiler(config);
+    const expConfig = await common.util.normalizeArguments(
```

Comment:
suggestion: would it be possible to move `normalizeArugments` call to an outer scope, and use the following here:
```ts
const expConfig = extend({},DEFAULT_CONFIG, config);
```

where `DEFAULT_CONFIG` could be:
```ts
const DEFAULT_CONFIG = await common.util.normalizeArguments(null, internalConfigParams);
Object.freeze(DEFAULT_CONFIG);
```

Reply:
It doesn't seem to be possible. 
I don't know why, but when I do this, I get a lot of:
```
     TypeError: Attempted to wrap instance which is already wrapped
```

Reply:
[patch.patch.zip](https://github.com/GoogleCloudPlatform/cloud-profiler-nodejs/files/1962425/patch.patch.zip)

Reply:
I think the reason it doesn't work is because mocha probably doesn't support `describe` blocks to be async.

I would suggest a patch along the lines of:

```diff
diff --git a/ts/test/test-init-config.ts b/ts/test/test-init-config.ts
index 4082ad588..27af2c09d 100644
--- a/ts/test/test-init-config.ts
+++ b/ts/test/test-init-config.ts
@@ -33,7 +33,23 @@ describe('createProfiler', async () => {
   let metadataStub: sinon.SinonStub|undefined;
   let startStub: sinon.SinonStub;
 
-  before(() => {
+  const internalConfigParams = {
+    timeIntervalMicros: 1000,
+    heapIntervalBytes: 512 * 1024,
+    heapMaxStackDepth: 64,
+    initialBackoffMillis: 1000 * 60,
+    backoffCapMillis: 60 * 60 * 1000,
+    backoffMultiplier: 1.3,
+    serverBackoffCapMillis: 2147483647,
+    localProfilingPeriodMillis: 1000,
+    localTimeDurationMillis: 1000,
+    localLogPeriodMillis: 10000,
+    baseApiUrl: 'https://cloudprofiler.googleapis.com/v2',
+  };
+  let defaultConfig : typeof internalConfigParams;
+
+  before(async () => {
+    defaultConfig = await common.util.normalizeArguments(null, extend({}, internalConfigParams));
     startStub = sinon.stub(v8HeapProfiler, 'startSamplingHeapProfiler');
     savedEnv = process.env;
   });
@@ -55,19 +71,8 @@ describe('createProfiler', async () => {
     startStub.restore();
   });
 
-  const internalConfigParams = {
-    timeIntervalMicros: 1000,
-    heapIntervalBytes: 512 * 1024,
-    heapMaxStackDepth: 64,
-    initialBackoffMillis: 1000 * 60,
-    backoffCapMillis: 60 * 60 * 1000,
-    backoffMultiplier: 1.3,
-    serverBackoffCapMillis: 2147483647,
-    localProfilingPeriodMillis: 1000,
-    localTimeDurationMillis: 1000,
-    localLogPeriodMillis: 10000,
-    baseApiUrl: 'https://cloudprofiler.googleapis.com/v2',
-  };
+
+
 
   it('should not modify specified fields when not on GCE', async () => {
     metadataStub = sinon.stub(gcpMetadata, 'instance')
@@ -83,8 +88,9 @@ describe('createProfiler', async () => {
       projectId: 'fake-projectId'
     };
     const profiler: Profiler = await createProfiler(config);
-    const expConfig = await common.util.normalizeArguments(
-        null, extend({}, internalConfigParams, config));
+    // const expConfig = await common.util.normalizeArguments(
+    //     null, extend({}, internalConfigParams, config));
+    const expConfig = Object.assign({}, defaultConfig, config);
     assert.deepEqual(profiler.config, expConfig);
   });
 

```

Reply:
Done.